src/main/java/com/thinkful/noteful/auth/WebSecurityConfig.java

```diff
@@ -0,0 +1,65 @@
+package com.thinkful.noteful.auth;
+
+import static com.thinkful.noteful.auth.SecurityConstants.SIGN_UP_URL;
+
+import com.thinkful.noteful.users.UserRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.CorsConfigurationSource;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+  private UserDetailsService userDetailsService;
+  private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+  public WebSecurityConfig(
+        UserDetailsService userDetailsService, 
+        BCryptPasswordEncoder bCryptPasswordEncoder) {
+    this.userDetailsService = userDetailsService;
+    this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+  }
+
+  @Override
+  protected void configure(HttpSecurity httpSecurity) throws Exception {
+    httpSecurity
+      .cors()
+      .and()
+      .csrf()
+      .disable()
+      .authorizeRequests()
+      .antMatchers(HttpMethod.POST, SIGN_UP_URL).permitAll()
```

Comment:
Wasn't sure if this was completed or if I'm missing something but wanted to verify that the SIGN_UP_URL permits user creation and sign up?

![sign up v2](https://user-images.githubusercontent.com/10905039/39965048-8abe45f4-5646-11e8-8657-8d10574c7e4c.PNG)

Reply:
Yes, this is done, you should be able to create a new user with the sign up url. I have examples in the README.md file.